src/functionaltests/java/com/ericsson/ei/query/FreestyleQueryAggregatedObjectsTestSteps.java

```diff
@@ -0,0 +1,149 @@
+    private String aggrCollectionName;
+    
+    @Value("${missedNotificationCollectionName}")
+    private String missedNotificationCollectionName;
+    
+    @Value("${missedNotificationDataBaseName}")
+    private String missedNotificationDatabaseName;
+
+    private String aggrObj;
+    
+    private ObjectMapper objMapper;
+    
+    public FreestyleQueryAggregatedObjectsTestSteps() {
+        objMapper = new ObjectMapper();
+
+        try {
+            aggrObj = FileUtils.readFileToString(new File(aggregatedObjJsonPath), "UTF-8");
+
+        } catch (IOException e) {
+            LOGGER.error("Failed to open test json files for test. Error message\n: " + e.getMessage());
+            e.printStackTrace();
+        }
+    }
+    
+    @Given("^Aggregated object2 is created$")
+    public void aggregated_object_is_created() throws Throwable {
+        LOGGER.debug("Creating aggregated object in MongoDb");
+        assertEquals(true, createDocumentInMongoDb(eiDatabaseName, aggrCollectionName, aggrObj));
+    }
+
+    @Then("^Perform valid freestyle query on newly created Aggregated object$")
+    public void perform_valid_query_on_newly_created_aggregated_object() throws Throwable {
+        final String expectedAggrId = "6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43";
+        final String entryPoint = "/query";
+        final String queryAggrObj = "{\"criteria\" :{\"aggregatedObject.id\" : \"" + expectedAggrId + "\", \"aggregatedObject.gav.groupId\" : \"com.mycompany.myproduct\"}}";
+
+        LOGGER.debug("Freestyle querying for the AggregatedObject with criteria: " + queryAggrObj);
+        mvcResult = mockMvc.perform(get(entryPoint)
+                .param("request", queryAggrObj)
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+        LOGGER.debug("Response of /query RestApi, Status Code: " + mvcResult.getResponse().getStatus() +
+                           "\nResponse: " + mvcResult.getResponse().getContentAsString());
+        
+        JsonNode jsonNodeResult = objMapper.readValue(mvcResult.getResponse().getContentAsString(), JsonNode.class);
+        JsonNode aggrObjResponse = objMapper.readValue(jsonNodeResult.get(0).get("aggregatedObject").toString(), JsonNode.class); 
+
+        String actualAggrObjId = aggrObjResponse.get("id").asText();
+        LOGGER.debug("AggregatedObject id from Response: " + actualAggrObjId);
+   
+        
+        assertEquals(HttpStatus.OK.toString(), Integer.toString(mvcResult.getResponse().getStatus()));
+        assertEquals("Failed to compare actual Aggregated Object Id:\n" + actualAggrObjId
+                + "\nwith expected Aggregated Object Id:\n" + expectedAggrId,
+                expectedAggrId, actualAggrObjId);
+    }
+    
+    @And("^Perform an invalid freesyle query on Aggregated object$")
+    public void perform_invalid_query_on_newly_created_aggregated_object() throws Throwable {
```

Comment:
add this step to the other steps file. Too much overhead to have a runner only for the steps here. This new runner will slow down testing since setting up embedded mongo db takes too much time.